app/src/androidTest/java/org/mozilla/fenix/ui/robots/BrowserRobot.kt

```diff
@@ -550,6 +550,94 @@ class BrowserRobot {
         }
     }
 
+    fun fillAndSubmitLoginCredentials(userName: String, password: String) {
+
+        val userNameTextBox =
+            mDevice.findObject(
+                UiSelector()
+                    .index(0)
+                    .resourceId("username")
+                    .className("android.widget.EditText")
+                    .packageName("$packageName")
+            )
+
+        userNameTextBox.click()
+        userNameTextBox.setText(userName)
+
+        val passwordTextBox =
+            mDevice.findObject(
+                UiSelector()
+                    .index(1)
+                    .resourceId("password")
+                    .className("android.widget.EditText")
+                    .packageName("$packageName")
+            )
+
+        passwordTextBox.click()
+        passwordTextBox.setText(password)
+
+        val submitLoginButton =
+            mDevice.findObject(
+                UiSelector()
+                    .index(2)
+                    .resourceId("submit")
+                    .textContains("Submit Query")
+                    .className("android.widget.Button")
+                    .packageName("$packageName")
+            )
+
+        submitLoginButton.click()
+        mDevice.waitForIdle()
+    }
+
+    fun verifyPrefilledLoginCredentials(userName: String, shortcutTitle: String) {
+
+        mDevice.waitForIdle()
+
+        val userNameTextBox =
+            mDevice.findObject(
+                UiSelector()
+                    .index(0)
+                    .resourceId("username")
+                    .className("android.widget.EditText")
+                    .packageName("$packageName")
+                    .text(userName)
+            )
+
+        val submitLoginButton =
+            mDevice.findObject(
+                UiSelector()
+                    .index(2)
+                    .resourceId("submit")
+                    .textContains("Submit Query")
+                    .className("android.widget.Button")
+                    .packageName("$packageName")
+            )
+
+        submitLoginButton.click()
+
+        var currentTries = 0
+        while (currentTries++ < 3) {
+            try {
+                assertTrue(userNameTextBox.waitForExists(waitingTime))
+                break
+            } catch (e: AssertionError) {
```

Comment:
Is this necessary? Can you try this with `RetryTestRule`?

Reply:
Yes it is, the RetryTestRule will not work.
I need to check the pre-filled in the PWA, the RetryTestRule takes me back to Fenix.